api/src/main/java/net/kyori/adventure/resource/ResourcePackRequest.java

```diff
@@ -191,6 +196,16 @@ interface Builder extends AbstractBuilder<ResourcePackRequest>, ResourcePackRequ
     @Override
     @NotNull ResourcePackRequest build();
 
+    /**
+     * Builds, computing a hash based on the provided URL.
+     *
+     * <p>The hash computation will perform a network request asynchronously, exposing the built request via the returned future.</p>
+     *
+     * @return a future providing the new resource pack request
+     * @since 4.15.0
+     */
+    @NotNull CompletableFuture<ResourcePackRequest> computeHashAndBuild();
```

Comment:
Any use for providing your own executor optionally?

Reply:
Sure, that'd make sense I suppose.